src/actions/sources.js

```diff
@@ -186,7 +188,11 @@ export function selectSource(id: string, options: SelectSourceOptions = {}) {
     let source = getSource(getState(), id);
 
     if (!source) {
-      return;
+      // If there is no source we deselect the current selected source
+      return dispatch({
+        type: constants.SELECT_SOURCE,
+        source: { id, url: null }
+      });
```

Comment:
Yeah, i totally agree! tot about this too, but didn't want to rock the boat yet! :)  having more focused actions that do specific things would allow for flexibility and less isolated scenario bugs!